### ISSUE TITLE
[question] checkpointing philosophy

### ISSUE THREAD

**Issue by jrjbertram** [hill-a/stable-baselines#87]
Hi, breaking this topic out into a separate issue from the thread of one of the other issues.

One of the things I was hoping to find was a way to do "checkpointing" which is incrementally capturing the state of the training session periodically so that if the server (or docker container) restarts, training starts over at the last checkpoint.  It could also be used I guess to continue training where you left off if you think additional training may have a benefit.

This is related to saving and restoring your model, which is mostly supported from what I can see from the docs and code.  It's not clear to me yet if the model load loads "all" state such that I can keep training without issues, or if the load is only suitable for evaluation where I would run the trained model in order to get an evaluation score... say without any exploration enabled.

So for this issue, I'm asking if you already have a solution in mind for how to approach this problem of not losing data past a checkpoint.  Assuming we fix the tensorboard restart issue, if we just call train in a loop... maybe that's all that's needed?   Or is there some enhancement that would be required?

Thanks!
- Josh.

**Comment by araffin** [comment]
Hello again,
Thank you for clarifying the issue.
So, from what I remember (that was @hill-a who mainly worked on that), if you call train multiple times (or after saving/loading), it will works. In fact, I used that feature to train models on Atari games on colab for 1e7 timesteps (because it was too long to do it in a single run).

The only thing that I'm not sure is if the state of the optimizer (and current learning rate) is saved or not.

>in order to get an evaluation score.

To evaluate your agent, you should use `predict` method rather than `train` method.
An example: https://github.com/araffin/rl-baselines-zoo/blob/master/enjoy.py
